packages/react-components/src/components/OnboardingChecklist/OnboardingChecklist.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+
+import { CheckCircle, ChevronDown } from '@livechat/design-system-icons';
+import cx from 'clsx';
+
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { Heading, Text } from '../Typography';
+
+import { CheckListItem } from './components';
+import { IOnboardingChecklistProps } from './types';
+
+import styles from './OnboardingChecklist.module.scss';
+
+const baseClass = 'onboarding-checklist';
+const COMPLETE_CONTAINER_EIGHT = 96;
+
+export const OnboardingChecklist: React.FC<IOnboardingChecklistProps> = ({
+  activeId,
+  checkedId,
+  title,
+  titleLabel,
+  items,
+  onActiveChange,
+  placeholderClassName,
+  isCompleted = false,
+  completeItem,
+  className,
+}) => {
+  const [complete, setComplete] = React.useState(isCompleted);
+  const [isOpen, setIsOpen] = React.useState(true);
+  const [currentContainerHeight, setCurrentContainerHeight] = React.useState<
+    number | undefined
+  >(undefined);
+  const containerRef = React.useRef<HTMLDivElement>(null);
+
+  const handleButtonClick = () => {
+    setIsOpen((prev) => !prev);
+  };
+
+  React.useEffect(() => {
+    const delay = completeItem?.delay || 1500;
+    const container = containerRef.current;
+
+    if (isCompleted && container) {
+      const currentHeight = container.offsetHeight;
+      setCurrentContainerHeight(currentHeight);
+
+      setTimeout(() => {
+        setComplete(true);
+        setIsOpen(false);
+      }, delay);
+    }
+  }, [isCompleted]);
+
+  return (
+    <div
+      ref={containerRef}
+      className={cx(
+        styles[baseClass],
+        {
+          [styles[`${baseClass}--complete`]]: complete && !isOpen,
+        },
+        className
+      )}
+      style={{
+        height:
+          !complete || isOpen
+            ? currentContainerHeight
+            : COMPLETE_CONTAINER_EIGHT,
```

Comment:
Will the completeStep always have the same height(COMPLETE_CONTAINER_EIGHT)? Because data from completeItem is dynamic